modules/configuration/src/main/java/org/apache/ignite/internal/configuration/asm/AbstractAsmGenerator.java

```diff
@@ -64,17 +64,20 @@ abstract class AbstractAsmGenerator {
     /** Configuration schema class. */
     final Class<?> schemaClass;
 
-    /** Internal extensions of the configuration schema. */
-    final Set<Class<?>> internalExtensions;
+    /** Extensions of the configuration schema (including internal ones). */
+    final Set<Class<?>> allConfigurationExtensions;
 
     /** Polymorphic extensions of the configuration schema. */
     final Set<Class<?>> polymorphicExtensions;
 
     /** Fields of the schema class. */
     final List<Field> schemaFields;
 
+    /** Fields of extensions of the configuration schema. */
+    final Collection<Field> extensionFields;
```

Comment:
It would be necessary to indicate that public extensions and do not forget about the javaDoc.